src/include/c.h

```diff
@@ -1211,6 +1211,20 @@ typedef union PGAlignedXLogBlock
 	((underlying_type) (expr))
 #endif
 
+#if defined __has_attribute
+	#if __has_attribute(fallthrough)
+		#define FALL_THROUGH __attribute__((fallthrough));
+	#else
+		#define FALL_THROUGH
+	#endif
+#else
+	/*
+	 * This is a fallback option for those compilers which don't support
+	 * "fallthrough" attribute. If that's the case, then do nothing.
+	 */
+	#define FALL_THROUGH
```

Comment:
Nitpick: We'll be using this from now on, so maybe make `FALL_THROUGH` macros look like a keyword similar to `break;` and `[[fallthrough]];` attribute to make it slightly more pleasant to look at? (`while` loop is to make semicolon required)
```suggestion
	#define fallthrough do {} while (0)
```

Reply:
it is tradition to write macros in uppercase. What happens without `do {} while (0)`? The semicolon will be required, because we use compiler with fallthrough attribute on CI

Reply:
> it is tradition to write macros in uppercase

There are exceptions for convenience, for example `elog()` and many more. My point is that `fallthrough;` can look like a keyword, and do opposite of `break;`.

> What happens without do {} while (0)?

So the `#else` macro definition requires a semicolon.

For the first definition, we may remove semicolon from the end of it to require a semicolon too:
```
#define fallthrough __attribute__((fallthrough))
```

Reply:
I would rather stick to a convention when we write macros in uppercase

Reply:
Sure then. Maybe still require a semicolon by doing:
```suggestion
	#define FALL_THROUGH do {} while (0)
```
and removing a semicolon from the attribute definition?

Reply:
I looked over the way macros are used in gp and now I would actually agree with you that they try to stylize them as functions/keywords. It depends, but maybe your proposition is suitable

Reply:
Looks like using `fallthrough` causes some conflicts with other macros. How about `fallthru` or just `through`?